subprocess/subtree_oom.go

```diff
@@ -0,0 +1,180 @@
+package subprocess
+
+import (
+	"os/exec"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type OOMTracker struct {
+	WasOOMKilled bool  `json:"was_oom_killed"`
+	IsSudo       bool  `json:"is_sudo"`
+	Pids         []int `json:"pids"`
+}
+
+func NewOOMTracker() *OOMTracker {
+	oomTracker := OOMTracker{}
+	return &oomTracker
+}
+
+func (o *OOMTracker) Clear() error {
+	return o.clear()
+}
+
+func (o *OOMTracker) Check() error {
+	return o.check()
+}
+
+func (o *OOMTracker) clear() error {
+	var err error
+	switch os := runtime.GOOS; os {
+	case "linux":
+		err = o.clearLinux()
```

Comment:
`return errors.Wrap(o.clearLinux() "problem clearing Linux")`

My reasoning here is that if you return `errors.Wrap`, it's better to do it on the line with the error, so that whoever is debugging the failure can go right there.